lib/spack/spack/stage.py

```diff
@@ -752,7 +752,8 @@ def purge():
 
 
 def get_checksums_for_versions(
-        url_dict, name, first_stage_function=None, keep_stage=False):
+        url_dict, name, first_stage_function=None, keep_stage=False,
+        package=None):
```

Comment:
A couple requests:

* Instead of passing `package`, can you just pass `fetch_options` (`spack create` might benefit from this in the future)?
* Can the docstring for that be slightly more verbose. E.g. "fetch_options (dict): used to initialize URLFetchStrategy (including connection timeouts and cookies)"

Also, flake is complaining.

Reply:
> * Instead of passing `package`, can you just pass `fetch_options` (`spack create` might benefit from this in the future)?

That makes more sense. Thanks!

> * Can the docstring for that be slightly more verbose. E.g. "fetch_options (dict): used to initialize URLFetchStrategy (including connection timeouts and cookies)"

Done.

> Also, flake is complaining.

Oops, fixed.